src/main/java/org/opensearch/flowframework/rest/RestProvisionWorkflowAction.java

```diff
@@ -68,24 +73,43 @@ public List<Route> routes() {
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         String workflowId = request.param(WORKFLOW_ID);
+        // Get any other params from path
+        Map<String, String> params = request.params()
+            .keySet()
+            .stream()
+            .filter(k -> !WORKFLOW_ID.equals(k))
+            .collect(Collectors.toMap(Function.identity(), request::param));
         try {
+            // If body is included get any params from body
+            if (request.hasContent()) {
+                try (XContentParser parser = request.contentParser()) {
+                    ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                    while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+                        String key = parser.currentName();
+                        if (params.containsKey(key)) {
+                            throw new FlowFrameworkException("Duplicate key " + key, RestStatus.BAD_REQUEST);
+                        }
+                        if (parser.nextToken() != XContentParser.Token.VALUE_STRING) {
```

Comment:
Because they are eventually passed to the `ProvisionWorkflowTransportAction` in the `WorkflowRequest`.  Since it uses the transport protocol, it uses `Writeable` implementation (`writeTo()` and `StreamInput` constructor) to pass the parameter map to the provisioning action.  

In theory we could do any `Writeable` implementation and the streams [do detect a finite number of simple subtitutions](https://github.com/opensearch-project/OpenSearch/blob/247e2eec46930e2eaec970b1acfb2f7d18e3e7a9/libs/core/src/main/java/org/opensearch/core/common/io/stream/StreamOutput.java#L795-L812) but that's really hard to try to document to users (even I don't understand it fully).  

We can always start with String and make it more flexible in the future if we get requests, but it's too hard to cover (and document) all the nuances of what is and is not writeable.  It's why JSON was invented and, fun enough, JSON is a string, which is where we'd end up parsing this on the other end anyway!  That's how we are returning the `Template` object in the get workflow API because making that whole object `Writable` is a lot harder than just sending a string over transport.

Reply:
> We can always start with String and make it more flexible in the future if we get requests,

It's fine to start with String and then increment later. For now, do you also want to include type `VALUE_BOOLEAN`?

Reply:
> For now, do you also want to include type VALUE_BOOLEAN?

No, because then I'd have to modify the `Map<String,String>` currently passed in the `WorkflowRequest` to a `Map<String, SomeOtherClassThatCanBeBothStringAndBoolean>`.  Which is pretty much just `Object`. :)

If I expanded it at all I'd use the same structure as the Settings API for a reference.   There are individual `Setting<T>` objects of various types, but the `Settings` class itself maintains a `String, String` map, or a `String, Object` map where the only acceptable object is a nested map.   The `Setting` objects themselves do the work of parsing themselves when you extract the (string) value on the other end of transport.

So the choices are:
1. Just specify a string, and deal with all the type parsing internally in the appropriate workflow step, e.g., specify port as "1234" and parse it to an int inside the step (which is what I do in the HttpHost step).  So if a user wants a boolean they just put "true".
2. Do a whole bunch of parsing before serializing for transport which just turns it into strings anyway and then parses it out on the other side, but we'd have to add a lot more overhead on transport to tell it HOW to parse.

Reply:
Dealing with parsing internally sounds good.